### PR TITLE
Core: Added ABORT() macro to prevent the usage of ASSERT(false) as a quick hack to crash the core misusing assert

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
-  add ABORT macros
##### Target branch(es):
- [x] master

### CODE REVIEW

File: src/common/Debugging/Errors.h [hunk 1]
```diff
@@ -13,26 +13,44 @@ namespace Trinity
...
 
+    DECLSPEC_NORETURN void AbortHandler(int sigval) ATTR_NORETURN;
+
 } // namespace Trinity
 
-#define WPAssert(cond) do { if (!(cond)) Trinity::Assert(__FILE__, __LINE__, __FUNCTION__, #cond); } while (0)
-#define WPFatal(cond, msg) do { if (!(cond)) Trinity::Fatal(__FILE__, __LINE__, __FUNCTION__, (msg)); } while (0)
-#define WPError(cond, msg) do { if (!(cond)) Trinity::Error(__FILE__, __LINE__, __FUNCTION__, (msg)); } while (0)
-#define WPWarning(cond, msg) do { if (!(cond)) Trinity::Warning(__FILE__, __LINE__, __FUNCTION__, (msg)); } while (0)
+#if COMPILER == COMPILER_MICROSOFT
```
Comment (FrancescoBorzi) [inline src/common/Debugging/Errors.h]: I guess we should test also the build on Windows?

Comment (Viste) [reply src/common/Debugging/Errors.h]: yes, but is should be fine.

Comment (vhiperdev) [reply src/common/Debugging/Errors.h]: VS 2017 Building Successfully

**Comment by Viste** [comment]
@milestorme fuck you stupid sucker :)))

**Comment by Viste** [comment]
so what about test review or something?

**Comment by Stoabrogga** [comment]
I'm sorry, but I can't do this because I don't know what this is about, why we need this or what this affects. I'm not good at C++.

**Comment by Viste** [comment]
> I'm sorry, but I can't do this because I don't know what this is about, why we need this or what this affects. I'm not good at C++.

`to prevent the usage of ASSERT(false) as a quick hack to crash the core`

**Comment by Stoabrogga** [comment]
Yes, I read that, but I don't understand why it is bad to use ASSERT(false). You also added some changes to Errors.cpp and Errors.h that I also don't understand.

**Comment by Stoabrogga** [comment]
As I told, I'm quite a noob at C++, sorry.

**Comment by Viste** [comment]
For better formatting and ace replace :)

**Comment by pklloveyou** [comment]
I'll use it on my server.

**Comment by FrancescoBorzi** [comment]
@pklloveyou that will be enough, thanks

**Comment by pklloveyou** [comment]
The server was running for 24 hours with no errors found

**Comment by Winfidonarleyan** [comment]
Test in win at VS 16.2. Perfect work
```
C:\Users\Kargatum\Desktop\D\Core\Azeroth>authserver.exe
AzerothCore rev. 24ba6797449c+ 2019-09-07 06:40:16 +0300 (ass branch) (Win64, RelWithDebInfo) (authserver)
<Ctrl-C> to stop.

   █████╗ ███████╗███████╗██████╗  ██████╗ ████████╗██╗  ██╗
  ██╔══██╗╚══███╔╝██╔════╝██╔══██╗██╔═══██╗╚══██╔══╝██║  ██║
  ███████║  ███╔╝ █████╗  ██████╔╝██║   ██║   ██║   ███████║
  ██╔══██║ ███╔╝  ██╔══╝  ██╔══██╗██║   ██║   ██║   ██╔══██║
  ██║  ██║███████╗███████╗██║  ██║╚██████╔╝   ██║   ██║  ██║
  ╚═╝  ╚═╝╚══════╝╚══════╝╚═╝  ╚═╝ ╚═════╝    ╚═╝   ╚═╝  ╚═╝
                                ██████╗ ██████╗ ██████╗ ███████╗
                                ██╔════╝██╔═══██╗██╔══██╗██╔═══╝
                                ██║     ██║   ██║██████╔╝█████╗
                                ██║     ██║   ██║██╔══██╗██╔══╝
                                ╚██████╗╚██████╔╝██║  ██║███████╗
                                 ╚═════╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝

          AzerothCore 3.3.5a  -  www.azerothcore.org

Using configuration file authserver.conf.

C:\Git\azerothcore-wotlk\src\server\authserver\Main.cpp:131 in ace_main_i ABORTED.
```

File: src/common/Debugging/Errors.cpp [hunk 2]
```diff
@@ -8,24 +8,45 @@
 
 #include <ace/Stack_Trace.h>
```
Comment (Winfidonarleyan) [inline src/common/Debugging/Errors.cpp]: maybe del include ace?

Comment (Viste) [reply src/common/Debugging/Errors.cpp]: Yes

**Pull request merged**